### PR TITLE
feat: server js fallthrough to page if handler doesn't exist

### LINKED ISSUE #10102: Prerendering with +server.js blocks cURL requests
### Describe the bug

When attempting to cURL a preview build of SvelteKit that uses `prerender` within a `+server.js` file a `405 Method not allowed` is returned. However when providing an extra "Accept" header `200 OK` is returned, and when using `+page.server.js` instead a `200 OK` is returned.

Example repository linked below.

### Reproduction

https://github.com/dextermb/svelte-kit-prerender-405-server-js

### Logs

_No response_

### System Info

```Shell
npx envinfo --system --binaries --browsers --npmPackages "{svelte,@sveltejs/*,vite}"
Need to install the following packages:
  envinfo@7.8.1
Ok to proceed? (y) y

  System:
    OS: macOS 13.4
    CPU: (12) arm64 Apple M2 Max
    Memory: 79.95 MB / 32.00 GB
    Shell: 5.9 - /bin/zsh
  Binaries:
    Node: 18.15.0 - ~/.nvm/versions/node/v18.15.0/bin/node
    npm: 9.5.0 - ~/.nvm/versions/node/v18.15.0/bin/npm
  Browsers:
    Chrome: 113.0.5672.126
    Firefox: 110.0.1
    Safari: 16.5
```

### Severity

annoyance

### Additional Information

_No response_

### PR DESCRIPTION
**Pull request by gtm-nayan**
closes #10102

### Please don't delete this checklist! Before submitting the PR, please make sure you do the following:
- [x] It's really useful if your PR references an issue where it is discussed ahead of time. In many cases, features are absent for a reason. For large changes, please create an RFC: https://github.com/sveltejs/rfcs
- [ ] This message body should clearly illustrate what problems it solves.
- [ ] Ideally, include a test that fails without this PR but passes with it.

### Tests
- [ ] Run the tests with `pnpm test` and lint the project with `pnpm lint` and `pnpm check`

### Changesets
- [ ] If your PR makes a change that should be noted in one or more packages' changelogs, generate a changeset by running `pnpm changeset` and following the prompts. Changesets that add features should be `minor` and those that fix bugs should be `patch`. Please prefix changeset messages with `feat:`, `fix:`, or `chore:`.

### CODE REVIEW

**Review by dummdidumm — CHANGES_REQUESTED** [review]
Needs to merge master in, master contains some changes which adjusts `response` after the if/else chain which would no longer apply.

File: packages/kit/src/runtime/server/respond.js [hunk 1]
```diff
@@ -392,16 +389,26 @@ export async function respond(request, options, manifest, state) {
 						trailing_slash ?? 'never'
 					);
 				} else if (route.endpoint && (!route.page || is_endpoint_request(event))) {
-					response = await render_endpoint(event, await route.endpoint(), state);
+					const endpoint = await route.endpoint();
+					const response = await render_endpoint(event, endpoint, state);
+
+					// If endpoint doesn't have a handler for these methods, fall back to the page
+					if (
+						response.status === 405 &&
```
Comment (benmccann) [inline packages/kit/src/runtime/server/respond.js:397]: did you mean for this to be `405`? based on the comment I would have expected `404`

Comment (s3812497) [reply packages/kit/src/runtime/server/respond.js:397]: Yeah, I think it's 405. Kit responds with `405 method not allowed` when the method handler isn't found.

**Comment by benmccann** [comment]
@gtm-nayan it looks like this PR still needs a rebase

**Comment by dummdidumm** [comment]
Through #11256 the underlying issue of conflicting routes was fixed, and as such this PR is no longer necessary - closing.

**Pull request closed without merging**